src/psyclone/parse/module_info.py

```diff
@@ -324,6 +325,22 @@ def get_symbol(self, name: str) -> Union[Symbol, None]:
         except KeyError:
             return None
 
+    def get_routine_by_name(
```

Comment:
This is a duplication of `Container.find_routine_psyir()` and `Container.resolve_routine()`. Can this be removed and that functionality used instead (from wherever you call this method).

Reply:
Still there?

Reply:
In fact, a grep reveals that this is only ever called from a test.